source/m3_parse.c

```diff
@@ -549,10 +549,10 @@ _   (Read_u32 (& version, & pos, end));
         u8 section;
 _       (ReadLEB_u7 (& section, & pos, end));
 
-        if (section > previousSection or            // from the spec: sections must appear in order
-            section == 0 or                         // custom section
-            (section == 12 and section == 9) or     // if present, DataCount goes after Element
-            (section == 10 and section == 12))      // and before Code
+        if (section > previousSection or                    // from the spec: sections must appear in order
+            section == 0 or                                 // custom section
+            (section == 12 and previousSection == 9) or     // if present, DataCount goes after Element
```

Comment:
This is never true because at this point `section <= previousSection`.

Reply:
:thinking: I think it should be OK because of the `or` right?

Reply:
I also think this condition might not be 100% correct. From what I understand section 12 (data count) **must** come before section 10 (code). So the sequence `11, 12` is currently allowed but I'm not sure if it should.